### PR TITLE
Move test data out of mypy/

### PR DESCRIPTION
**Pull request by gnprice**
This makes it easier to grep through our codebase to find things
quickly without getting noise from matches in our tests --
after all, our test data naturally has a lot of Python code,
just like mypy itself.  #compilerworldproblems

Specifically, move what was mypy/test/data/ inside a new directory
test-data/.  While we're at it, tuck samples/ and stdlib-samples/
under test-data/ too.  This is probably still not the perfect layout
(e.g., does stdlib-samples/ need to be special, or can it just be more
samples organized into a directory under samples/?), but with all the
file moves in this commit I wanted to keep the necessary logic changes
to an absolute minimum for ease of reading and reviewing.

The `-M` flag to `git show/diff/log` is highly recommended when
looking at this commit.

### CODE REVIEW

File: mypy/test/testparse.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 """Tests for the mypy parser
-
-Test case descriptions are in files test/data/parse[-errors].test."""
+"""
```
Comment (gvanrossum) [inline mypy/test/testparse.py]: Proper docstring style is actually to have the close """ on the same line *iff* it's a single line. So it was wrong, and it's still wrong. :-)

**Comment by gvanrossum** [comment]
LGTM except for the docstring nit! I've wanted this for so long...

(I also want all test code out of the mypy package, but that may have to wait and it's also a bit debatable.)

**Comment by gvanrossum** [comment]
Are you sure samples/ is test data? I thought it was examples. Though
stdlib-samples is inded test data.

On Monday, June 6, 2016, Greg Price <notifications@github.com> wrote:

> This makes it easier to grep through our codebase to find things
> quickly without getting noise from matches in our tests --
> after all, our test data naturally has a lot of Python code,
> just like mypy itself. #compilerworldproblems
>
> Specifically, move what was mypy/test/data/ inside a new directory
> test-data/. While we're at it, tuck samples/ and stdlib-samples/
> under test-data/ too. This is probably still not the perfect layout
> (e.g., does stdlib-samples/ need to be special, or can it just be more
> samples organized into a directory under samples/?), but with all the
> file moves in this commit I wanted to keep the necessary logic changes
> to an absolute minimum for ease of reading and reviewing.
>
> The -M flag to git show/diff/log is highly recommended when
> looking at this commit.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/python/mypy/pull/1667
> Commit Summary
>
>    - Move test data out of mypy/
>
> File Changes
>
>    - *M* mypy/build.py
>    <https://github.com/python/mypy/pull/1667/files#diff-0> (6)
>    - *M* mypy/test/config.py
>    <https://github.com/python/mypy/pull/1667/files#diff-1> (5)
>    - *M* mypy/test/helpers.py
>    <https://github.com/python/mypy/pull/1667/files#diff-2> (3)
>    - *M* mypy/test/testparse.py
>    <https://github.com/python/mypy/pull/1667/files#diff-3> (3)
>    - *M* runtests.py
>    <https://github.com/python/mypy/pull/1667/files#diff-4> (8)
>    - *M* setup.cfg <https://github.com/python/mypy/pull/1667/files#diff-5>
>    (2)
>    - *R* test-data/samples/bottles.py
>    <https://github.com/python/mypy/pull/1667/files#diff-6> (0)
>    - *R* test-data/samples/class.py
>    <https://github.com/python/mypy/pull/1667/files#diff-7> (0)
>    - *R* test-data/samples/cmdline.py
>    <https://github.com/python/mypy/pull/1667/files#diff-8> (0)
>    - *R* test-data/samples/codec/example.py
>    <https://github.com/python/mypy/pull/1667/files#diff-9> (0)
>    - *R* test-data/samples/codec/example_non_default_coding.py
>    <https://github.com/python/mypy/pull/1667/files#diff-10> (0)
>    - *R* test-data/samples/codec/example_utf8_bom.py
>    <https://github.com/python/mypy/pull/1667/files#diff-11> (0)
>    - *R* test-data/samples/dict.py
>    <https://github.com/python/mypy/pull/1667/files#diff-12> (0)
>    - *R* test-data/samples/fib.py
>    <https://github.com/python/mypy/pull/1667/files#diff-13> (0)
>    - *R* test-data/samples/files.py
>    <https://github.com/python/mypy/pull/1667/files#diff-14> (0)
>    - *R* test-data/samples/for.py
>    <https://github.com/python/mypy/pull/1667/files#diff-15> (0)
>    - *R* test-data/samples/generators.py
>    <https://github.com/python/mypy/pull/1667/files#diff-16> (0)
>    - *R* test-data/samples/greet.py
>    <https://github.com/python/mypy/pull/1667/files#diff-17> (0)
>    - *R* test-data/samples/guess.py
>    <https://github.com/python/mypy/pull/1667/files#diff-18> (0)
>    - *R* test-data/samples/hello.py
>    <https://github.com/python/mypy/pull/1667/files#diff-19> (0)
>    - *R* test-data/samples/input.py
>    <https://github.com/python/mypy/pull/1667/files#diff-20> (0)
>    - *R* test-data/samples/itertool.py
>    <https://github.com/python/mypy/pull/1667/files#diff-21> (0)
>    - *R* test-data/samples/readme.txt
>    <https://github.com/python/mypy/pull/1667/files#diff-22> (0)
>    - *R* test-data/samples/regexp.py
>    <https://github.com/python/mypy/pull/1667/files#diff-23> (0)
>    - *R* test-data/stdlib-samples/3.2/base64.py
>    <https://github.com/python/mypy/pull/1667/files#diff-24> (0)
>    - *R* test-data/stdlib-samples/3.2/fnmatch.py
>    <https://github.com/python/mypy/pull/1667/files#diff-25> (0)
>    - *R* test-data/stdlib-samples/3.2/genericpath.py
>    <https://github.com/python/mypy/pull/1667/files#diff-26> (0)
>    - *R* test-data/stdlib-samples/3.2/getopt.py
>    <https://github.com/python/mypy/pull/1667/files#diff-27> (0)
>    - *R* test-data/stdlib-samples/3.2/glob.py
>    <https://github.com/python/mypy/pull/1667/files#diff-28> (0)
>    - *R* test-data/stdlib-samples/3.2/incomplete/logging/__init__.py
>    <https://github.com/python/mypy/pull/1667/files#diff-29> (0)
>    - *R* test-data/stdlib-samples/3.2/incomplete/urllib/__init__.py
>    <https://github.com/python/mypy/pull/1667/files#diff-30> (0)
>    - *R* test-data/stdlib-samples/3.2/incomplete/urllib/parse.py
>    <https://github.com/python/mypy/pull/1667/files#diff-31> (0)
>    - *R* test-data/stdlib-samples/3.2/posixpath.py
>    <https://github.com/python/mypy/pull/1667/files#diff-32> (0)
>    - *R* test-data/stdlib-samples/3.2/pprint.py
>    <https://github.com/python/mypy/pull/1667/files#diff-33> (0)
>    - *R* test-data/stdlib-samples/3.2/random.py
>    <https://github.com/python/mypy/pull/1667/files#diff-34> (0)
>    - *R* test-data/stdlib-samples/3.2/shutil.py
>    <https://github.com/python/mypy/pull/1667/files#diff-35> (0)
>    - *R* test-data/stdlib-samples/3.2/subprocess.py
>    <https://github.com/python/mypy/pull/1667/files#diff-36> (0)
>    - *R* test-data/stdlib-samples/3.2/tempfile.py
>    <https://github.com/python/mypy/pull/1667/files#diff-37> (0)
>    - *R* test-data/stdlib-samples/3.2/test/__init__.py
>    <https://github.com/python/mypy/pull/1667/files#diff-38> (0)
>    - *R* test-data/stdlib-samples/3.2/test/randv2_32.pck
>    <https://github.com/python/mypy/pull/1667/files#diff-39> (0)
>    - *R* test-data/stdlib-samples/3.2/test/randv2_64.pck
>    <https://github.com/python/mypy/pull/1667/files#diff-40> (0)
>    - *R* test-data/stdlib-samples/3.2/test/randv3.pck
>    <https://github.com/python/mypy/pull/1667/files#diff-41> (0)
>    - *R* test-data/stdlib-samples/3.2/test/subprocessdata/fd_status.py
>    <https://github.com/python/mypy/pull/1667/files#diff-42> (0)
>    - *R* test-data/stdlib-samples/3.2/test/subprocessdata/input_reader.py
>    <https://github.com/python/mypy/pull/1667/files#diff-43> (0)
>    - *R* test-data/stdlib-samples/3.2/test/subprocessdata/qcat.py
>    <https://github.com/python/mypy/pull/1667/files#diff-44> (0)
>    - *R* test-data/stdlib-samples/3.2/test/subprocessdata/qgrep.py
>    <https://github.com/python/mypy/pull/1667/files#diff-45> (0)
>    - *R*
>    test-data/stdlib-samples/3.2/test/subprocessdata/sigchild_ignore.py
>    <https://github.com/python/mypy/pull/1667/files#diff-46> (0)
>    - *R* test-data/stdlib-samples/3.2/test/support.py
>    <https://github.com/python/mypy/pull/1667/files#diff-47> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_base64.py
>    <https://github.com/python/mypy/pull/1667/files#diff-48> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_fnmatch.py
>    <https://github.com/python/mypy/pull/1667/files#diff-49> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_genericpath.py
>    <https://github.com/python/mypy/pull/1667/files#diff-50> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_getopt.py
>    <https://github.com/python/mypy/pull/1667/files#diff-51> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_glob.py
>    <https://github.com/python/mypy/pull/1667/files#diff-52> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_posixpath.py
>    <https://github.com/python/mypy/pull/1667/files#diff-53> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_pprint.py
>    <https://github.com/python/mypy/pull/1667/files#diff-54> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_random.py
>    <https://github.com/python/mypy/pull/1667/files#diff-55> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_set.py
>    <https://github.com/python/mypy/pull/1667/files#diff-56> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_shutil.py
>    <https://github.com/python/mypy/pull/1667/files#diff-57> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_subprocess.py
>    <https://github.com/python/mypy/pull/1667/files#diff-58> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_tempfile.py
>    <https://github.com/python/mypy/pull/1667/files#diff-59> (0)
>    - *R* test-data/stdlib-samples/3.2/test/test_textwrap.py
>    <https://github.com/python/mypy/pull/1667/files#diff-60> (0)
>    - *R* test-data/stdlib-samples/3.2/test/tf_inherit_check.py
>    <https://github.com/python/mypy/pull/1667/files#diff-61> (0)
>    - *R* test-data/stdlib-samples/3.2/textwrap.py
>    <https://github.com/python/mypy/pull/1667/files#diff-62> (0)
>    - *R* test-data/unit/check-abstract.test
>    <https://github.com/python/mypy/pull/1667/files#diff-63> (0)
>    - *R* test-data/unit/check-basic.test
>    <https://github.com/python/mypy/pull/1667/files#diff-64> (0)
>    - *R* test-data/unit/check-bound.test
>    <https://github.com/python/mypy/pull/1667/files#diff-65> (0)
>    - *R* test-data/unit/check-classes.test
>    <https://github.com/python/mypy/pull/1667/files#diff-66> (0)
>    - *R* test-data/unit/check-dynamic-typing.test
>    <https://github.com/python/mypy/pull/1667/files#diff-67> (0)
>    - *R* test-data/unit/check-expressions.test
>    <https://github.com/python/mypy/pull/1667/files#diff-68> (0)
>    - *R* test-data/unit/check-fastparse.test
>    <https://github.com/python/mypy/pull/1667/files#diff-69> (0)
>    - *R* test-data/unit/check-flags.test
>    <https://github.com/python/mypy/pull/1667/files#diff-70> (0)
>    - *R* test-data/unit/check-functions.test
>    <https://github.com/python/mypy/pull/1667/files#diff-71> (0)
>    - *R* test-data/unit/check-generic-subtyping.test
>    <https://github.com/python/mypy/pull/1667/files#diff-72> (0)
>    - *R* test-data/unit/check-generics.test
>    <https://github.com/python/mypy/pull/1667/files#diff-73> (0)
>    - *R* test-data/unit/check-ignore.test
>    <https://github.com/python/mypy/pull/1667/files#diff-74> (0)
>    - *R* test-data/unit/check-incremental.test
>    <https://github.com/python/mypy/pull/1667/files#diff-75> (0)
>    - *R* test-data/unit/check-inference-context.test
>    <https://github.com/python/mypy/pull/1667/files#diff-76> (0)
>    - *R* test-data/unit/check-inference.test
>    <https://github.com/python/mypy/pull/1667/files#diff-77> (0)
>    - *R* test-data/unit/check-isinstance.test
>    <https://github.com/python/mypy/pull/1667/files#diff-78> (0)
>    - *R* test-data/unit/check-kwargs.test
>    <https://github.com/python/mypy/pull/1667/files#diff-79> (0)
>    - *R* test-data/unit/check-lists.test
>    <https://github.com/python/mypy/pull/1667/files#diff-80> (0)
>    - *R* test-data/unit/check-modules.test
>    <https://github.com/python/mypy/pull/1667/files#diff-81> (0)
>    - *R* test-data/unit/check-multiple-inheritance.test
>    <https://github.com/python/mypy/pull/1667/files#diff-82> (0)
>    - *R* test-data/unit/check-namedtuple.test
>    <https://github.com/python/mypy/pull/1667/files#diff-83> (0)
>    - *R* test-data/unit/check-overloading.test
>    <https://github.com/python/mypy/pull/1667/files#diff-84> (0)
>    - *R* test-data/unit/check-python2.test
>    <https://github.com/python/mypy/pull/1667/files#diff-85> (0)
>    - *R* test-data/unit/check-semanal-error.test
>    <https://github.com/python/mypy/pull/1667/files#diff-86> (0)
>    - *R* test-data/unit/check-statements.test
>    <https://github.com/python/mypy/pull/1667/files#diff-87> (0)
>    - *R* test-data/unit/check-super.test
>    <https://github.com/python/mypy/pull/1667/files#diff-88> (0)
>    - *R* test-data/unit/check-tuples.test
>    <https://github.com/python/mypy/pull/1667/files#diff-89> (0)
>    - *R* test-data/unit/check-type-aliases.test
>    <https://github.com/python/mypy/pull/1667/files#diff-90> (0)
>    - *R* test-data/unit/check-type-checks.test
>    <https://github.com/python/mypy/pull/1667/files#diff-91> (0)
>    - *R* test-data/unit/check-type-promotion.test
>    <https://github.com/python/mypy/pull/1667/files#diff-92> (0)
>    - *R* test-data/unit/check-typevar-values.test
>    <https://github.com/python/mypy/pull/1667/files#diff-93> (0)
>    - *R* test-data/unit/check-unions.test
>    <https://github.com/python/mypy/pull/1667/files#diff-94> (0)
>    - *R* test-data/unit/check-unreachable-code.test
>    <https://github.com/python/mypy/pull/1667/files#diff-95> (0)
>    - *R* test-data/unit/check-unsupported.test
>    <https://github.com/python/mypy/pull/1667/files#diff-96> (0)
>    - *R* test-data/unit/check-varargs.test
>    <https://github.com/python/mypy/pull/1667/files#diff-97> (0)
>    - *R* test-data/unit/check-weak-typing.test
>    <https://github.com/python/mypy/pull/1667/files#diff-98> (0)
>    - *R* test-data/unit/cmdline.test
>    <https://github.com/python/mypy/pull/1667/files#diff-99> (0)
>    - *R* test-data/unit/fixtures/__new__.py
>    <https://github.com/python/mypy/pull/1667/files#diff-100> (0)
>    - *R* test-data/unit/fixtures/alias.py
>    <https://github.com/python/mypy/pull/1667/files#diff-101> (0)
>    - *R* test-data/unit/fixtures/args.py
>    <https://github.com/python/mypy/pull/1667/files#diff-102> (0)
>    - *R* test-data/unit/fixtures/bool-simple.py
>    <https://github.com/python/mypy/pull/1667/files#diff-103> (0)
>    - *R* test-data/unit/fixtures/bool.py
>    <https://github.com/python/mypy/pull/1667/files#diff-104> (0)
>    - *R* test-data/unit/fixtures/classmethod.py
>    <https://github.com/python/mypy/pull/1667/files#diff-105> (0)
>    - *R* test-data/unit/fixtures/complex.py
>    <https://github.com/python/mypy/pull/1667/files#diff-106> (0)
>    - *R* test-data/unit/fixtures/dict.py
>    <https://github.com/python/mypy/pull/1667/files#diff-107> (0)
>    - *R* test-data/unit/fixtures/exception.py
>    <https://github.com/python/mypy/pull/1667/files#diff-108> (0)
>    - *R* test-data/unit/fixtures/for.py
>    <https://github.com/python/mypy/pull/1667/files#diff-109> (0)
>    - *R* test-data/unit/fixtures/function.py
>    <https://github.com/python/mypy/pull/1667/files#diff-110> (0)
>    - *R* test-data/unit/fixtures/isinstance.py
>    <https://github.com/python/mypy/pull/1667/files#diff-111> (0)
>    - *R* test-data/unit/fixtures/isinstancelist.py
>    <https://github.com/python/mypy/pull/1667/files#diff-112> (0)
>    - *R* test-data/unit/fixtures/list.py
>    <https://github.com/python/mypy/pull/1667/files#diff-113> (0)
>    - *R* test-data/unit/fixtures/module.py
>    <https://github.com/python/mypy/pull/1667/files#diff-114> (0)
>    - *R* test-data/unit/fixtures/module_all.py
>    <https://github.com/python/mypy/pull/1667/files#diff-115> (0)
>    - *R* test-data/unit/fixtures/ops.py
>    <https://github.com/python/mypy/pull/1667/files#diff-116> (0)
>    - *R* test-data/unit/fixtures/primitives.py
>    <https://github.com/python/mypy/pull/1667/files#diff-117> (0)
>    - *R* test-data/unit/fixtures/property.py
>    <https://github.com/python/mypy/pull/1667/files#diff-118> (0)
>    - *R* test-data/unit/fixtures/python2.py
>    <https://github.com/python/mypy/pull/1667/files#diff-119> (0)
>    - *R* test-data/unit/fixtures/set.py
>    <https://github.com/python/mypy/pull/1667/files#diff-120> (0)
>    - *R* test-data/unit/fixtures/slice.py
>    <https://github.com/python/mypy/pull/1667/files#diff-121> (0)
>    - *R* test-data/unit/fixtures/staticmethod.py
>    <https://github.com/python/mypy/pull/1667/files#diff-122> (0)
>    - *R* test-data/unit/fixtures/transform.py
>    <https://github.com/python/mypy/pull/1667/files#diff-123> (0)
>    - *R* test-data/unit/fixtures/tuple-simple.py
>    <https://github.com/python/mypy/pull/1667/files#diff-124> (0)
>    - *R* test-data/unit/fixtures/tuple.py
>    <https://github.com/python/mypy/pull/1667/files#diff-125> (0)
>    - *R* test-data/unit/fixtures/union.py
>    <https://github.com/python/mypy/pull/1667/files#diff-126> (0)
>    - *R* test-data/unit/lib-stub/__builtin__.py
>    <https://github.com/python/mypy/pull/1667/files#diff-127> (0)
>    - *R* test-data/unit/lib-stub/abc.py
>    <https://github.com/python/mypy/pull/1667/files#diff-128> (0)
>    - *R* test-data/unit/lib-stub/builtins.py
>    <https://github.com/python/mypy/pull/1667/files#diff-129> (0)
>    - *R* test-data/unit/lib-stub/collections.py
>    <https://github.com/python/mypy/pull/1667/files#diff-130> (0)
>    - *R* test-data/unit/lib-stub/typing.py
>    <https://github.com/python/mypy/pull/1667/files#diff-131> (0)
>    - *R* test-data/unit/parse-errors.test
>    <https://github.com/python/mypy/pull/1667/files#diff-132> (0)
>    - *R* test-data/unit/parse-python2.test
>    <https://github.com/python/mypy/pull/1667/files#diff-133> (0)
>    - *R* test-data/unit/parse.test
>    <https://github.com/python/mypy/pull/1667/files#diff-134> (0)
>    - *R* test-data/unit/python2eval.test
>    <https://github.com/python/mypy/pull/1667/files#diff-135> (0)
>    - *R* test-data/unit/pythoneval-asyncio.test
>    <https://github.com/python/mypy/pull/1667/files#diff-136> (0)
>    - *R* test-data/unit/pythoneval-enum.test
>    <https://github.com/python/mypy/pull/1667/files#diff-137> (0)
>    - *R* test-data/unit/pythoneval.test
>    <https://github.com/python/mypy/pull/1667/files#diff-138> (0)
>    - *R* test-data/unit/semanal-abstractclasses.test
>    <https://github.com/python/mypy/pull/1667/files#diff-139> (0)
>    - *R* test-data/unit/semanal-basic.test
>    <https://github.com/python/mypy/pull/1667/files#diff-140> (0)
>    - *R* test-data/unit/semanal-classes.test
>    <https://github.com/python/mypy/pull/1667/files#diff-141> (0)
>    - *R* test-data/unit/semanal-errors.test
>    <https://github.com/python/mypy/pull/1667/files#diff-142> (0)
>    - *R* test-data/unit/semanal-expressions.test
>    <https://github.com/python/mypy/pull/1667/files#diff-143> (0)
>    - *R* test-data/unit/semanal-modules.test
>    <https://github.com/python/mypy/pull/1667/files#diff-144> (0)
>    - *R* test-data/unit/semanal-namedtuple.test
>    <https://github.com/python/mypy/pull/1667/files#diff-145> (0)
>    - *R* test-data/unit/semanal-python2.test
>    <https://github.com/python/mypy/pull/1667/files#diff-146> (0)
>    - *R* test-data/unit/semanal-statements.test
>    <https://github.com/python/mypy/pull/1667/files#diff-147> (0)
>    - *R* test-data/unit/semanal-symtable.test
>    <https://github.com/python/mypy/pull/1667/files#diff-148> (0)
>    - *R* test-data/unit/semanal-typealiases.test
>    <https://github.com/python/mypy/pull/1667/files#diff-149> (0)
>    - *R* test-data/unit/semanal-typeinfo.test
>    <https://github.com/python/mypy/pull/1667/files#diff-150> (0)
>    - *R* test-data/unit/semanal-types.test
>    <https://github.com/python/mypy/pull/1667/files#diff-151> (0)
>    - *R* test-data/unit/stubgen.test
>    <https://github.com/python/mypy/pull/1667/files#diff-152> (0)
>    - *R* test-data/unit/typexport-basic.test
>    <https://github.com/python/mypy/pull/1667/files#diff-153> (0)
>
> Patch Links:
>
>    - https://github.com/python/mypy/pull/1667.patch
>    - https://github.com/python/mypy/pull/1667.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/python/mypy/pull/1667>, or mute the thread
> <https://github.com/notifications/unsubscribe/ACwrMlKSCCT9RA62ySR9K3lIKHo1lP1dks5qJMZ7gaJpZM4Ivdsb>
> .
>


-- 
--Guido (mobile)

**Comment by gnprice** [comment]
Pushed a fix to that docstring. :-)

We do use `samples/` as test data -- see `add_samples` in runtests.py. Which isn't to say it doesn't also serve as examples to look at. My guess is that Jukka originally wrote them as examples of code in the Mypy language.

**Comment by gvanrossum** [comment]
OK. LGTM.

--Guido (mobile)

**Comment by gnprice** [comment]
(We're going to hold this a couple of days for the 0.4.2 release, after which I'll rebase and merge.)

**Comment by JukkaL** [comment]
This is good to rebase and merge now!

**Pull request merged**